public/static/app/common/Rounder.js

```diff
@@ -0,0 +1,62 @@
+/* Relative Error Rounder
+ *   It convert a number to string and always keep rounding
+ *   error less tham some predefined value (default 0.1)
+ *   For number >=1000 it just applies AngularJS number filter
+ *   TODO should not require AngularJS Number filter
+ * Functions:
+ * - get(options) - returns a rounder functions
+ *   options:
+ *   - rre - Relative Rounding Error (default 0.1)
+ *   - trailingZeros - Show trailing zeros (default false)
+ *
+ * Examples:
+ * let r = Rounder.get()
+ * r(0.01234) // -> '0.012'
+ * ----
+ * let r = Rounder.get({rre: 0.01, showTrailingZeros: true})
+ * r(0.01) // -> '0.0100'
+*/
+mciModule.factory('Rounder', function(numberFilter) {
+  const defaults = {
+    rre: 0.1,
+    trailingZeros: false,
+  }
+
+  const expBreakPoint = 2
+
+  function get(options) {
+    options = Object.assign({}, defaults, options || {})
+    let takeN = Math.abs(Math.log10(options.rre)) + 1
+
+    return function(v) {
+      let absV = Math.abs(v)
+      let exponent = Math.floor(Math.log10(absV))
+      // FIXME may produce less precise results for values > 1000
+      //       and rre < 0.0001
+      if (exponent > expBreakPoint) { return numberFilter(v, 0) }
+
+      // Perorm conversion to string and rounding
+      let mantissaS = Math.round(absV / Math.pow(10, exponent - takeN + 1))
```

Comment:
Why not just do number.toPrecision?

Reply:
`toPrecision` could't be used as a drop in replacement here, because I use "Rounding half away from zero" strategy https://en.wikipedia.org/wiki/Rounding#Round_half_away_from_zero (I consider it as more intuitive), and `toPrecision` rounds towards zero. Don't say it matters a lot, though. But doesn't seem to be changed either.
The other point is that this code gives more control over the logic, because it is explicit.
And the last one, `toPrecision` + wrapping code is just a little bit shorter and simpler than, this code, which is not enough to convince me change anything here.